opentracing-spring-web-handler-interceptor/src/main/java/io/opentracing/contrib/spring/web/servlet/TracingHandlerInterceptor.java

```diff
@@ -0,0 +1,103 @@
+package io.opentracing.contrib.spring.web.servlet;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.web.servlet.filter.HttpServletHeadersExtractAdapter;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+import io.opentracing.propagation.Format;
+import io.opentracing.tag.Tags;
+
+/**
+ * Tracing handler interceptor for spring web. It creates a new span for an incoming request.
+ * Created span can be accessed via {@link HttpServletRequest#getAttribute(String)} with name
+ * {@link TracingHandlerInterceptor#ACTIVE_SPAN}.
+ *
+ * @author Pavol Loffay
+ */
+public class TracingHandlerInterceptor extends HandlerInterceptorAdapter {
+
+    /**
+     * Used as a name of {@link HttpServletRequest#setAttribute(String, Object)} to inject current server span
+     */
+    private static final String ACTIVE_SPAN = TracingHandlerInterceptor.class.getName() + ".currentSpan";
+
+    private Tracer tracer;
+    private List<SpanDecorator> decorators;
+    private TracingDecision tracingDecision;
+
+    /**
+     * @param tracer tracer
+     * @param decorators span decorators
+     * @param tracingDecision tracing decision
+     */
+    public TracingHandlerInterceptor(Tracer tracer, List<SpanDecorator> decorators, TracingDecision tracingDecision) {
+        this.tracer = tracer;
+        this.decorators = new ArrayList<>(decorators);
+        this.tracingDecision = tracingDecision;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, Object handler)
+            throws Exception {
+
+        Span serverSpan = (Span) httpServletRequest.getAttribute(TracingFilter.ACTIVE_SPAN);
+        SpanWrapper spanWrapper = (SpanWrapper) httpServletRequest.getAttribute(ACTIVE_SPAN);
+
+        if (serverSpan != null && spanWrapper == null) {
+            for (SpanDecorator decorator: decorators) {
+                decorator.onPreHandle(httpServletRequest, handler, serverSpan);
+            }
+            spanWrapper = new SpanWrapper(serverSpan, true);
+            httpServletRequest.setAttribute(ACTIVE_SPAN, spanWrapper);
+        }else if (spanWrapper == null && httpServletRequest.getAttribute(ACTIVE_SPAN) == null &&
```

Comment:
`spanWrapper == null && httpServletRequest.getAttribute(ACTIVE_SPAN) == null`

I guess one of above is sufficient as spanWrapper == httpServletRequest.getAttribute(ACTIVE_SPAN)